test/SadTest.cc

```diff
@@ -64,31 +66,23 @@ extern "C" void ext_eight_sad_calculation_32x32_64x64_c(
     uint32_t *p_best_mv64x64, uint32_t mv, uint32_t p_sad32x32[4][8]);
 
 namespace {
-#define MAX_BLOCK_SIZE (128 * 256)
+#define MAX_SRC_BLOCK_SIZE (256 * 256)
+#define MAX_REF_BLOCK_SIZE (720 * 480)
 typedef std::tuple<int, int> SadSize;
 typedef enum { REF_MAX, SRC_MAX, RANDOM, UNALIGN } TestPattern;
 typedef enum { BUF_MAX, BUF_MIN, BUF_RANDOM } SADPattern;
-SadSize TEST_SAD_SIZES[] = {SadSize(64, 64),
-                            SadSize(64, 32),
-                            SadSize(32, 64),
-                            SadSize(32, 32),
-                            SadSize(32, 16),
-                            SadSize(16, 32),
-                            SadSize(16, 16),
-                            SadSize(16, 8),
-                            SadSize(8, 16),
-                            SadSize(8, 8),
-                            SadSize(8, 4),
-                            SadSize(4, 8),
-                            SadSize(4, 16),
-                            SadSize(16, 4),
-                            SadSize(8, 32),
-                            SadSize(32, 8),
-                            SadSize(16, 64),
-                            SadSize(64, 16)};
+SadSize TEST_SAD_SIZES[] = {
+    SadSize(64, 64), SadSize(64, 32), SadSize(32, 64), SadSize(32, 32),
+    SadSize(32, 16), SadSize(16, 32), SadSize(16, 16), SadSize(16, 8),
+    SadSize(8, 16),  SadSize(8, 8),   SadSize(8, 4),   SadSize(4, 4),
+    SadSize(4, 8),   SadSize(4, 16),  SadSize(16, 4),  SadSize(8, 32),
+    SadSize(32, 8),  SadSize(16, 64), SadSize(64, 16), SadSize(24, 24),
+    SadSize(24, 16), SadSize(16, 24), SadSize(24, 8),  SadSize(8, 24),
```

Comment:
I think we should leave some comment to explain why we add 24, 48, since normal block size could not be 24 or 48.